### PR TITLE
add ecs capabilities

### PR DESCRIPTION
**Pull request by jcalderin**
Added:
* ECS cluster
* ECS capacity providers
* ECS task definition

### CODE REVIEW

File: tools/gen-providers.ts [hunk 1]
```diff
@@ -2925,6 +2959,100 @@ const ssoadmin_permission_set_inline_policy: RecordDecl = {
...
+    optionalField('execute_command_configuration', recordType(ecs_cluster_configuration_execute_command_configuration)),
+  ]
+}
+
+const ecs_cluster_setting: RecordDecl = {
+  name: 'ecs_cluster_setting',
+  fields: [
+    requiredField('name', STRING),
+    requiredField('value', enumType(['enabled', 'disabled'])),
+
```
Comment (bacek) [inline tools/gen-providers.ts]: formatting

File: tools/gen-providers.ts [hunk 2]
```diff
@@ -2925,6 +2959,100 @@ const ssoadmin_permission_set_inline_policy: RecordDecl = {
...
+}
+
+const ecs_cluster: RecordDecl = {
+  name: 'ecs_cluster',
+  fields: [
+    optionalField('configuration', recordType(ecs_cluster_configuration)),
+    requiredField('name', STRING),
+    optionalField('setting', recordType(ecs_cluster_setting)),
+    optionalField('tags', TAGS_MAP),
+
```
Comment (bacek) [inline tools/gen-providers.ts]: formatting

**Review by bacek — CHANGES_REQUESTED** [review]

File: tools/gen-providers.ts (hunk 1 above)
Comment (jcalderin) [reply tools/gen-providers.ts]: mmm... i should have the pre-commit doing this 🤔

**Comment by bacek** [comment]
:+1:

**Review by bacek — APPROVED** [review]

**Pull request merged**